build.c

```diff
@@ -495,6 +521,15 @@ jobwork(struct job *j)
 	return false;
 }
 
+#if HAVE_TIOCGWINSZ
+static void
+sig_handler(int sig)
+{
+	if (sig == SIGWINCH)
+		ioctl(1, TIOCGWINSZ, &ws);
```

Comment:
I went with the first option and it seems to work.

Reply:
If you don't think it's a problem to go with the third method, I think that might be preferable just because of the complexity in the code that watching for SIGWINCH has added compared to the one line it would be to check the width for each status printing?